users/nazu/keyrecords/process_records.c

```diff
@@ -0,0 +1,228 @@
+/* Copyright 2023 Tom Sennewald (@beelzebubi)
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "nazu.h"
+#include "version.h"
+
+/**
+ * @brief Keycode handler for keymaps
+ *
+ * This handles the keycodes at the keymap level, useful for keyboard specific customization
+ */
+__attribute__((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+__attribute__((weak)) bool process_record_secrets(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
+bool useTapDance = true;
+uint8_t mod_state;
+/**
+ * @brief Main user keycode handler
+ *
+ * This handles all of the keycodes for the user, including calling feature handlers.
+ *
+ * @param keycode Keycode from matrix
+ * @param record keyrecord_t data structure
+ * @return true Continue processing keycode and send to host
+ * @return false Stop process keycode and do not send to host
+ */
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+// #ifdef OLED_ENABLE
+//     process_record_user_oled(keycode, record);
+// #endif // OLED_ENABLE
+
+    // Store the current modifier state in the variable for later reference
+    mod_state = get_mods();
+    switch (keycode) {
+        case KC_QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+        case KC_COLEMAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            return false;
+        case KC_COLEMAK_DH:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK_DH);
+            }
+            return false;
+        case KC_SWAP:
+            keymap_config.raw = eeconfig_read_keymap();
+            if (record->event.pressed) {
+                keymap_config.swap_lalt_lgui = true;
+                eeconfig_update_keymap(keymap_config.raw);
+                clear_keyboard(); // clear to prevent stuck keys
+            }
+            return false;
+        case KC_NORM:
+            keymap_config.raw = eeconfig_read_keymap();
+            if (record->event.pressed) {
+                keymap_config.swap_lalt_lgui = false;
+                eeconfig_update_keymap(keymap_config.raw);
+                clear_keyboard(); // clear to prevent stuck keys
+            }
+            return false;
+        case KC_R2:
```

Comment:
Not sure that R1 and R2 are needed, as these should be handled by a normal KC_RALT keycode in the keymap (as these do get processed by the magic config.

Also, `keycode_config` and `mod_config` will process the inputted value and return the "correct" value.

Reply:
I tried to work with the magic config but I couldn't find a valid solution appropriate for my needs.
On the right side of the spacebar, I want GUI and ALT when I'm on Mac and ALT and CTRL. As far as I understood and tried the magic configs, this specific config wasn't achievable.